apiserver/facades/agent/uniter/uniter.go

```diff
@@ -1685,7 +1685,7 @@ func (u *UniterAPI) readOneRemoteSettings(ctx context.Context, canAccess common.
 		if err != nil {
 			return nil, internalerrors.Capture(err)
 		}
-		settings, err = u.relationService.GetRemoteRelationApplicationSettings(ctx, relUUID, appID)
+		settings, err = u.relationService.GetRelationApplicationSettings(ctx, relUUID, appID)
```

Comment:
`GetRelationApplicationSettings` is not the same as `GetRemoteRelationApplicationSettings`. To make it the same, you need to call GetRelationApplicationSettings with the application ID of the application on the other end of the relation, not the application for the given unit. Reread the facade method descriptions.

Reply:
Those names would make sense if that was what the facade did, but juju loves to mislead here :upside_down_face:. The facade actually calls this method with the remote application's ID. It already has both. 

See the full code here, we actually use the remote application tag to call this method:
https://github.com/juju/juju/blob/964c014aadd9a563e61fda009c2edc7b47ca1a42/apiserver/facades/agent/uniter/uniter.go#L1719-L1772